src/arena.h

```diff
@@ -0,0 +1,16 @@
+#ifndef ARENA_H
+#define ARENA_H
+#include <stddef.h>
+
+typedef struct {
+    size_t capacity;
+    size_t count;
+    char data[];
```

Comment:
```suggestion
    char *data;
```
Use a pointer member since cpp don't allow flexible array member.